src/csharp/Microsoft.Spark.Worker/Processor/BroadcastVariableProcessor.cs

```diff
@@ -48,8 +58,14 @@ internal BroadcastVariables Process(Stream stream)
                 {
                     if (broadcastVars.DecryptionServerNeeded)
                     {
-                        throw new NotImplementedException(
-                            "broadcastDecryptionServer is not implemented.");
+                        var readBid = SerDe.ReadInt64(socket.InputStream);
+                        if (bid != readBid)
+                        {
+                            throw new Exception($"Broadcast id {readBid} from encrypted stream" +
+                                $" does not " + $"match broadcast id {bid} from normal stream.");
```

Comment:
What is encrypted stream vs. "normal" stream? The stream we are reading from is encrypted? I don't see any decryption going on.

Reply:
Sorry, I didn't mean the stream is encrypted (you are right there is no decryption going on since we store value directly), but that the stream is from the socket connection to the encryption server on scala side, as opposed to the stream from the other socket that receives the rest of the payload. Please advice the best way to frame this exception message. Thank you.